### PR TITLE
Add GetMetadata() to s3util to get s3 metadata information

### PR DESCRIPTION
**Pull request by shuzhang1989**
to be used in scorpion to get md5 of objects

### CODE REVIEW

File: common/s3util.cpp [hunk 1]
```diff
@@ -124,6 +126,21 @@ GetObjectsResponse S3Util::getObjects(
...
 
+GetObjectMetadataResponse S3Util::getObjectMetadata(const string &key) {
+  HeadObjectRequest headObjectRequest;
+  headObjectRequest.SetBucket(bucket_);
+  headObjectRequest.SetKey(key);
+  auto headObjectResult = s3Client.HeadObject(headObjectRequest);
+  map<string, string> metadata;
+  if (headObjectResult.IsSuccess()) {
+    metadata = headObjectResult.GetResult().GetMetadata();
+    return GetObjectMetadataResponse(metadata, "");
```
Comment (chengren311) [inline common/s3util.cpp]: NIT: should we return something indicating the request is successful instead of an empty string?

Comment (shuzhang1989) [inline common/s3util.cpp]: No, i use str.empty() to check if its a good response or not.

Comment (jinruh) [inline common/s3util.cpp]: nit: std::move(metadata)

**Comment by newpoo** [comment]
please add std::move() as suggested by jinru

---

Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pinterest/rocksplicator/27#-:-Ke45HEJaxeFXCbuErmm:b96r7w9)*

**Pull request merged**